From 009450c7ad4d425fa5a10ac4bd6efbd25248d823 Mon Sep 17 00:00:00 2001
From: Brian Gladman <riemannic@gmail.com>
Date: Mon, 27 Jul 2020 17:01:46 +0100
Subject: [PATCH 41/79] minor update to VS2019 build

---
 Mkfiles/vs/yasm.sln | 9 ++++++---
 1 file changed, 6 insertions(+), 3 deletions(-)

diff --git a/Mkfiles/vs/yasm.sln b/Mkfiles/vs/yasm.sln
index 5d2dc952..8d3b03a4 100644
--- a/Mkfiles/vs/yasm.sln
+++ b/Mkfiles/vs/yasm.sln
@@ -1,11 +1,11 @@
 Microsoft Visual Studio Solution File, Format Version 12.00
-# Visual Studio 14
-VisualStudioVersion = 14.0.22823.1
+# Visual Studio Version 16
+VisualStudioVersion = 16.0.30320.27
 MinimumVisualStudioVersion = 10.0.40219.1
 Project("{2150E333-8FDC-42A3-9474-1A3956D46DE8}") = "Solution Items", "Solution Items", "{825AC694-358C-4D8D-92DE-33A2691978CE}"
 	ProjectSection(SolutionItems) = preProject
 		crt_secure_no_deprecate.vsprops = crt_secure_no_deprecate.vsprops
-		readme.vc14.txt = readme.vc14.txt
+		readme.vs.txt = readme.vs.txt
 	EndProjectSection
 EndProject
 Project("{8BC9CEB8-8B4A-11D0-8D11-00A0C91BC942}") = "libyasm", "libyasm\libyasm.vcxproj", "{29FE7874-1256-4AD6-B889-68E399DC9608}"
@@ -139,4 +139,7 @@ Global
 	GlobalSection(SolutionProperties) = preSolution
 		HideSolutionNode = FALSE
 	EndGlobalSection
+	GlobalSection(ExtensibilityGlobals) = postSolution
+		SolutionGuid = {BE3F09BB-91A9-474C-AAD9-4B4D43DCE89E}
+	EndGlobalSection
 EndGlobal
-- 
2.49.0

