From 2cd3bb50e256f5ed5f611ac611d25fe673f2cec3 Mon Sep 17 00:00:00 2001
From: Peter Johnson <johnson.peter@gmail.com>
Date: Fri, 30 Jun 2023 08:08:55 -0700
Subject: [PATCH 71/79] elf.c: Fix NULL deref on bad xsize expression (#234)

---
 modules/objfmts/elf/elf.c | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/modules/objfmts/elf/elf.c b/modules/objfmts/elf/elf.c
index 67fe3f26..1ee98bfd 100644
--- a/modules/objfmts/elf/elf.c
+++ b/modules/objfmts/elf/elf.c
@@ -482,15 +482,15 @@ elf_symtab_write_to_file(FILE *f, elf_symtab_head *symtab,
 
         /* get size (if specified); expr overrides stored integer */
         if (entry->xsize) {
-            size_intn = yasm_intnum_copy(
-                yasm_expr_get_intnum(&entry->xsize, 1));
-            if (!size_intn) {
+            yasm_intnum *intn = yasm_expr_get_intnum(&entry->xsize, 1);
+            if (!intn) {
                 yasm_error_set(YASM_ERROR_VALUE,
                                N_("size specifier not an integer expression"));
                 yasm_errwarn_propagate(errwarns, entry->xsize->line);
-            }
+            } else
+                size_intn = yasm_intnum_copy(intn);
         }
-        else
+        if (!size_intn)
             size_intn = yasm_intnum_create_uint(entry->size);
 
         /* get EQU value for constants */
-- 
2.49.0

